test/e2e/tests/weighted_backend.go

```diff
@@ -0,0 +1,225 @@
+
+				} else {
+					// all we need is the pod Name prefix
+					podNamePrefix := ExtractPodNamePrefix(podName)
+					weightMap[podNamePrefix]++
+				}
+			}
+
+			// Since we only route to pods with "infra-backend-v 1" and "infra-backend-v 2" prefixes
+			// So here we use fixed weight values
+			v1Weight := 9
+			v2Weight := 1
+
+			if !CalculateWeight(v1Weight, v2Weight, weightMap) {
+				t.Errorf("The actual traffic weights are not consistent with the expected routing weights")
+			}
+		})
+	},
+}
+
+var WeightCompleteRolloutTest = suite.ConformanceTest{
+	ShortName:   "WeightCompleteRollout",
+	Description: "Resource with Weight Backend enabled, and use the completing rollout policy weight setting",
+	Manifests:   []string{"testdata/weight-backend-completing-rollout.yaml"},
+	Test: func(t *testing.T, suite *suite.ConformanceTestSuite) {
+		t.Run("all backends receive the complete rollout weight of traffic", func(t *testing.T) {
+			ns := "gateway-conformance-infra"
+			weightEqualRoute := types.NamespacedName{Name: "weight-complete-rollout-http-route", Namespace: ns}
+			gwNN := types.NamespacedName{Name: "same-namespace", Namespace: ns}
+			gwAddr := kubernetes.GatewayAndHTTPRoutesMustBeAccepted(t, suite.Client, suite.TimeoutConfig,
+				suite.ControllerName,
+				kubernetes.NewGatewayRef(gwNN), weightEqualRoute)
+
+			expectedResponse := http.ExpectedResponse{
+				Request: http.Request{
+					Path: "/",
+				},
+				Response: http.Response{
+					StatusCode: 200,
+				},
+				Namespace: ns,
+			}
+			req := http.MakeRequest(t, &expectedResponse, gwAddr, "HTTP", "http")
+
+			weightMap := make(map[string]int)
+			for i := 0; i < 10; i++ {
+				cReq, cResp, err := suite.RoundTripper.CaptureRoundTrip(req)
+				if err != nil {
+					t.Errorf("failed to get expected response: %v", err)
+				}
+
+				if err := http.CompareRequest(t, &req, cReq, cResp, expectedResponse); err != nil {
+					t.Errorf("failed to compare request and response: %v", err)
+				}
+
+				podName := cReq.Pod
+				if len(podName) == 0 {
+					// it shouldn't be missing here
+					t.Errorf("failed to get pod header in response: %v", err)
+
+				} else {
+					// all we need is the pod Name prefix
+					podNamePrefix := ExtractPodNamePrefix(podName)
+					weightMap[podNamePrefix]++
+				}
+			}
+
+			// Since we only route to pods with "infra-backend-v 1" and "infra-backend-v 2" prefixes
+			// So here we use fixed weight values
+			v1Weight := 10
+			v2Weight := 0
+			if !CalculateWeight(v1Weight, v2Weight, weightMap) {
+				t.Errorf("The actual traffic weights are not consistent with the expected routing weights")
+			}
+		})
+	},
+}
+
+// CalculateWeight Calculate whether the expected weight is equal to the weight of the expected traffic
+func CalculateWeight(weight1, weight2 int, actualMap map[string]int) bool {
+
+	var count int
+	for _, v := range actualMap {
+		count += v
+	}
+	expectedV1 := float64(weight1) / float64(weight1+weight2) * float64(count)
```

Comment:
I think we should use `AlmostEquals` here, 10 requests on `1:1`, may result `4:6` in real world.

Reply:
Yes, this problem does occur. The actual traffic can only be infinitely close to the expected weight, but is not always guaranteed to be equal. (This is related to the cardinality of our request). Do you have a better solution? grateful!

Reply:
One solution is to increase the request base to 100 and give a deviation tolerance when calling the weight calculation.

Reply:
try istio's way?
```golang
func AlmostEquals(a, b, precision int) bool {
	upper := a + precision
	lower := a - precision
	if b < lower || b > upper {
		return false
	}
	return true
}
```

Reply:
It seems that istio's idea is very good, I will choose this method! 
Gratitude!